common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -165,7 +165,7 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
 
     DownloadCallback(int chunkIndex) throws IOException {
       this.targetFile = tempFileManager.createTempFile();
-      this.channel = Channels.newChannel(Files.newOutputStream(targetFile.toPath()));
+      this.channel = Channels.newChannel(new FileOutputStream(targetFile));
```

Comment:
Seems `Channels.newChannel` has an optimization for `FileOutputStream`: http://www.grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/java/nio/channels/Channels.java#424

Reply:
Here I think we can use `FileChannel.open` instead.